pages/api/zapbye.ts

```diff
@@ -0,0 +1,98 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import jsforce from 'jsforce';
+
+const CONSTANTS = {
+      SALESFORCE_EMAIL: process.env.SALESFORCE_EMAIL,
+      SALESFORCE_PASSWORD: process.env.SALESFORCE_PASSWORD,
+      SALESFORCE_SECURITY_TOKEN: process.env.SALESFORCE_SECURITY_TOKEN,
+      LEMLIST_API_KEY: process.env.LEMLIST_API_KEY,
+      SALESFORCE_TASK_SUBJECT: 'Lemlist Email',
+      SALESFORCE_LEMLISTTASK_ID: '0123k0000014TCyAAM',
+      SALESFORCE_STATUS: 'Completed',
+      SALESFORCE_TASK_SUBTYPE: 'Task',
+      SALESFORCE_SOBJECT_TYPE: 'Task',
+      LEMLIST_CAMPAIGN_FETCH_COOLDOWN_TIME_IN_MS: 30000,
+};
+
+let salesforceConnection;
+let lastCampaignCheckedTimestamp: number;
+let cachedCampaignsArray: Array<LemlistCampaign> = [];
+
+export default async (req: NextApiRequest, res: NextApiResponse): Promise<void> => {
+      const {
+            method,
+      } = req;
+
+      if (method != "POST") {
+            res.status(405).json({ error: "Method Not Allowed" });
+            return;
+      }
+
+      if(!salesforceConnection || !salesforceConnection.accessToken) {
+            console.log("Attempting to log in!");
+            salesforceConnection = new jsforce.Connection();
+            await salesforceConnection.login(CONSTANTS.SALESFORCE_EMAIL, CONSTANTS.SALESFORCE_PASSWORD + CONSTANTS.SALESFORCE_SECURITY_TOKEN, function(err, res) {
```

Comment:
@adobito I don't think you need the callback here. It's pretty common in javascript to have functions have both a **callback** and a **promise** handler, mostly because some apps haven't converted to use promise yet, it mostly came on the scene around the time of es5.

In this case, you likely don't want to pass an argument for callback, and instead use the *result* of the promise by assigning a value from the await.

Reply:
Following up, took a look at the [API](https://jsforce.github.io/jsforce/doc/Connection.html#login)
```suggestion
   try {
            const userInfo = await salesforceConnection.login(CONSTANTS.SALESFORCE_EMAIL, CONSTANTS.SALESFORCE_PASSWORD + CONSTANTS.SALESFORCE_SECURITY_TOKEN);
            ... //do whatever with userInfo
            } catch (err: any) => {
                ...
            }
```

Reply:
This might take a few mistakes for me to get used to, but thanks for the great example. Got it confirmed working.